### PR TITLE
Add acceptance tests to verify special character handling in timeout storages

### PR DESCRIPTION
**Pull request by timbussmann**
One test for using delayed retries and one for delayed sends.

ping @weralabaj

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/DelayedDelivery/When_using_special_characters_in_headers.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        {
+            Requires.TimeoutStorage();
+
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<EndpointHandlingDelayedMessages>(e => e
+                    .When(s =>
+                    {
+                        var options = new SendOptions();
+                        options.RouteToThisEndpoint();
+                        options.DelayDeliveryWith(TimeSpan.FromSeconds(3));
```
Comment (weralabaj) [inline src/NServiceBus.AcceptanceTests/DelayedDelivery/When_using_special_characters_in_headers.cs]: in this case will it touch the storage? or can it be in memory only?

Comment (weralabaj) [inline src/NServiceBus.AcceptanceTests/DelayedDelivery/When_using_special_characters_in_headers.cs]: in this case will it touch the storage? or can it be in memory only?

Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/DelayedDelivery/When_using_special_characters_in_headers.cs]: yes with that value it should go through the timeout storage (see https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/DelayedDelivery/TimeoutManager/StoreTimeoutBehavior.cs#L64)

**Comment by bording** [comment]
Rebased to fix merge conflicts, will merge once green since there were other approvals before the rebase.

**Pull request merged**